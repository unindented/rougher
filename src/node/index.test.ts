import { readFileSync } from "fs";
import { resolve } from "path";

import roughUp from ".";

const readFixture = (fixture: string) =>
  readFileSync(resolve(__dirname, "__fixtures__", fixture)).toString();

describe("node roughUp", () => {
  describe.each`
    name                          | fixture
    ${"a processing instruction"} | ${readFixture("processing-instruction.svg")}
  `("with $name", ({ fixture }: { fixture: string }) => {
    it("returns a roughed up version", () => {
      expect(roughUp(fixture)).toMatchSnapshot();
    });
  });
});
