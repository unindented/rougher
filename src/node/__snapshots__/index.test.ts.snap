// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node roughUp with a processing instruction returns a roughed up version 1`] = `
"<?xml-stylesheet href=\\"styles.css\\" type=\\"text/css\\"?><svg viewBox=\\"0 0 100 100\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <g><path d=\\"M4.271824857220054 3.2576828692108393 C25.138676139526073 4.068767916318029, 45.42620269954205 3.9021759740076956, 94.80876251123846 5.7751799914985895 M5.613966231234372 5.763947914354503 C26.73993713408709 5.396963874530047, 47.41608370374889 3.874028633069247, 95.38871649373323 4.733868996612728 M96.66412683390081 6.066399225965142 C94.33315986385568 28.624970427714292, 92.36240649214014 49.701983783394105, 94.93674220703542 93.48307580687106 M95.54342797864228 4.811957041732967 C95.43512335112318 34.972813142463565, 94.55234312703833 63.77384696993977, 94.2318487809971 95.57250751089305 M95.92896260879934 96.95408935286105 C74.25662709120661 95.48673654953018, 49.460899468511336 96.1443900601007, 6.234557947143912 96.34666658379138 M95.65541249979287 94.41677750181407 C63.0247921962291 94.13014254430308, 33.23688906338065 94.73722935179248, 4.4081691252067685 94.73184285592288 M5.455919584259391 94.6942517850548 C6.1764527195133265 70.67311355192214, 5.3274377816356715 46.79456298053264, 3.3789065089076757 5.196091482415795 M5.206786249764264 94.77906237076968 C3.8285125856287774 60.11709273606538, 3.804144150484353 27.934595261700437, 4.671614293940365 5.493582795374095\\" stroke=\\"#000\\" stroke-width=\\"1\\" fill=\\"none\\"/></g>
</svg>"
`;
